src/psyclone/psyir/nodes/node.py

```diff
@@ -1149,6 +1149,40 @@ def scope(self):
             "Unable to find the scope of node '{0}' as none of its ancestors "
             "are Container or Schedule nodes.".format(self))
 
+    def replace_with(self, node):
+        '''Replaces self with the supplied node in the PSyIR tree.
+
+        :param node: the node that will replace self in the PSyIR \
+            tree.
+        :type node: subclass of :py:class:`psyclone.psyir.nodes.node`
+
+        :raises TypeError: if the argument 'node' is not a Node.
+        :raises SymbolError: if this node does not have a parent.
```

Comment:
You meant GenerationError here and line below

Reply:
Oops, I was trying different options, one of which was SymbolError, and forgot to update the docstring.